src/core/server/plugins/plugins_service.ts

```diff
@@ -56,7 +56,7 @@ export interface PluginsServiceStart {
 export type PluginsServiceSetupDeps = InternalCoreSetup;
 
 /** @internal */
-export interface PluginsServiceStartDeps {} // eslint-disable-line @typescript-eslint/no-empty-interface
+export type PluginsServiceStartDeps = InternalCoreStart;
```

Comment:
I'm not totally convinced we want exact type equivalence here ?

In https://github.com/elastic/kibana/pull/51438/files#diff-725c498a267472754f01cff2baa3901b I explicitly defined the properties

```
- export interface PluginsServiceStartDeps {} // eslint-disable-line @typescript-eslint/no-empty-interface
+ export interface PluginsServiceStartDeps {
+   capabilities: InternalCoreStart['capabilities'];
+ }
```

But I'm not sure which one is preferable? I'm guessing we may not necessarily want alls internal core in the service deps? wdyt?

Reply:
I don't think it *needs* to be exactly equivalent, but I don't see the advantage to a more minimal typing, especially since the plugin service will probably require almost every API except the legacy ones which we'll remove later. The more strictly we type it, the more boilerplate it takes to introduce a new API. Although it makes dependencies explicit, I don't think having several layers of slightly different types makes the code more readable, gives us better guarantees or reduced bugs.

Reply:
Reduced boilerplate is a very valid argument